### ISSUE TITLE
Fargate module tries to reuse existing TaskDefinition instead of using the latest task definition

### ISSUE THREAD

**Issue by danbosnichill** [umotif-public/terraform-aws-ecs-fargate#40]
**What is the current behavior?**

I have both a Fargate Terraform module and a GitHub Action  deploying Fargate.  GitHub Action knows to updates the latest TaskDefinition.  The Fargate Terraform module continues to update the old version that it created.

```
aws-actions/amazon-ecs-render-task-definition@v1
aws-actions/amazon-ecs-deploy-task-definition@v1
```

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

I don't have a great way to show the repo.  We have a big, private repo.  I'm guessing this will be pretty easy to confirm by someone that knows this code.  I'm guessing this is working as implemented.

**What is the expected behavior?**

I'd expect this module to either reuse the latest TaskDefinition or create a new TaskDefinition.

**Software versions?**

6.1.0

**Comment by seanpascual** [comment]
Hi @danbosnichill

Just so that we can understand your use-case for this module, is the GitHub Action updating the same fargate ECS instance as this module?

If so, you are looking to have this module look for any external updates to the task definition and update / use that one rather than the last version that this module created?

Thanks,

Sean

**Comment by stackexpress-tanmaya** [comment]
@seanpascual yes. Can you please look at the PR and let me know if its suitable for the purpose.

**Comment by danbosnichill** [comment]
Yes.  @stackexpress-tanmaya and I work together.

**Comment by seanpascual** [comment]
Hi @danbosnichill @stackexpress-tanmaya I have a small change requested in the PR. Once that it changed I can merge and release this. :)

**Comment by stackexpress-tanmaya** [comment]
@seanpascual done. Please have a look when possible

**Comment by seanpascual** [comment]
Hey Both,

I've released this as 6.2.1

Thank you for your help in fixing this issue. 😄 Please advise if you run into any further issues.

Cheers,

Sean

**Comment by stackexpress-tanmaya** [comment]
Will do. Thanks for quick your guidance and merge!
